presets/ragengine/main.py

```diff
@@ -197,9 +224,21 @@ async def query_index(request: QueryRequest):
         llm_params = request.llm_params or {}  # Default to empty dict if no params provided
         rerank_params = request.rerank_params or {}  # Default to empty dict if no params provided
         
-        result = await rag_ops.query(
+        result_dict = await rag_ops.query(
             request.index_name, request.query, request.top_k, llm_params, rerank_params
         )
+        
+        result = QueryResponse(
+            response=result_dict["response"],
+            source_nodes=result_dict["source_nodes"], 
+            metadata=result_dict.get("metadata", {})
+        )
+        
+        # Get lowest score source
+        if result.source_nodes and result.response:
+            lowest_score_node = min(result.source_nodes, key=lambda x: x.score)
+            rag_lowest_source_score.observe(lowest_score_node.score)
```

Comment:
in addition to lowest score - observe
average - rag_avg_source_score.observe(avg_score)
optional - number of scores len(scores) - this is like number documents retrieved